server/@types/dto/index.d.ts

```diff
@@ -0,0 +1,217 @@
+declare module 'dto' {
+  /**
+   * Interface defining common reference and audit related properties that DTO types can inherit through extension.
+   */
+  interface ReferencedAndAuditable {
```

Comment:
I couldnt think of a better name, but basically what I'm trying to say is that any type that extends this becomes "referenced" and "audited" (perhaps `ReferencedAndAudited` is better?)

Reply:
I like it but personally I think `Auditable` is perfectly ok, regardless of the reference field.